fix(cli): ctrl c/ctrl d close cli when in dialogs

lifefloating opened the pull request:
## TLDR

Implements proper Ctrl+C exit handling with confirmation for all dialog types. Previously, Ctrl+C did not work consistently when dialogs were open. Now pressing Ctrl+C twice will reliably close dialogs with visual feedback on the first press.

## Dive Deeper

This PR addresses a critical UX issue where users could get stuck in dialogs without a reliable way to exit. The implementation:

- Adds a two-press Ctrl+C system for dialogs: first press shows warning, second press closes
- Implements consistent behavior across all dialog types (settings, editor, theme, privacy notice)
- Adds visual feedback "Press Ctrl+C again to exit" to guide users
- Prevents accidental exits during authentication flows
- Maintains existing Ctrl+C behavior for normal CLI operations
- Includes comprehensive test coverage for all dialog scenarios


## Reviewer Test Plan

1. **Test each dialog type:**
   - Open settings dialog (`/settings`) → Press Ctrl+C once → Verify warning appears → Press Ctrl+C again → Verify dialog closes
   - Open editor dialog (`/editor`) → Repeat above steps
   - Open theme dialog → Repeat above steps
   - Trigger privacy notice → Repeat above steps


## Testing Matrix

|          | 🍏  | 🪟  | 🐧  |
| -------- | --- | --- | --- |
| npm run  | ✅  | ❓  | ❓  |

## Linked issues / bugs

Fixes #8648

jacob314 reviewed (commented):
I would hope we can solve this a bit more simply with a single handler in AppContainer rather than having this logic in all the dialogs. If that is not possible we can go this way but I would prefer we refactor to achieve that. If that is not feasible, we need to add a base component for all dialogs so this logic does not get repeated.  Otherwise we will miss adding it next time we add a dialog.

lifefloating commented:
refactor and in a more simple way

Unified Exit Prompt Location

  - Before: Each dialog had its own "Press Ctrl+C again to exit" prompt
  - Now: All exit prompts are unified and displayed in App.tsx

 Removed Redundant Dialog Closing Logic

  - Before: Complex logic in AppContainer.tsx where the second Ctrl+C would check and close various dialogs one by one
  - Now: Deleted all specific dialog closing handling code

jacob314 commented on packages/cli/src/ui/AppContainer.tsx:
```diff
@@ -865,14 +865,28 @@ Logging in with Google... Please restart Gemini CLI to continue.
-        isFolderTrustDialogOpen ||
-        showPrivacyNotice;
-      if (anyDialogOpen) {
+      if (keyMatchers[Command.QUIT](key)) {
```
I pushed a commit simplifying this a little further. I don't think we needed to check for the dialogs at all.

jacob314 reviewed (approved):
![lgtm](https://i.imgur.com/JgNRalI.png)
